src/mame/nmk/nmk214.h

```diff
@@ -0,0 +1,38 @@
+// license:BSD-3-Clause
+// copyright-holders:Sergio Galiano
+#ifndef MAME_NMK_NMK214_H
+#define MAME_NMK_NMK214_H
+
+#pragma once
+
+
+class nmk214_device : public device_t
+{
+public:
+	nmk214_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	void set_mode(u8 mode) { m_mode = mode; }
+	void set_init_config(u8 init_config);
+	bool is_device_initialized() { return m_device_initialized; }
+	void set_input_address_bitswap(const u8 *input_address_bitswap) { m_input_address_bitswap = input_address_bitswap; }
+	
+	u16 decode_word(u32 addr, u16 data);
+	u8 decode_byte(u32 addr, u8 data);
+
+protected:
+	virtual void device_start() override;
+	virtual void device_reset() override;
+
+private:
+	u8 m_mode;                              // Operation mode. In practice, only LSB is used. Allowed values: 0 or 1. This value is hardwired for each NM214 out of 2 ones in the game board, each device having a different configuration than the other.
+	u8 m_init_config;                       // An init config out of 8 existing ones in NMK214 to be used while descrambling GFX. In practice, only lower 3 bits (From 0 to 2) are used, allowed values: 0 to 7. Bit 3 is used to match with operation mode and let the init config to be stored on the device or not
+	bool m_device_initialized;			    // Flag to mark the device already received the init configuration and it can perform descrambling from now on
+
+	const u8 *m_input_address_bitswap;      // Input address lines bitswap. These represents the way the 13 address lines on NMK214 (from A0 to A12) are hooked up externally related to the GFX ROMs address bus, that could be in different order/position
```

Comment:
It’s somewhat dangerous using a pointer for this – you’ll get no array bounds checking at compile time, etc.  Can you use `std::array<u8, 13>` or C array references or something to make it safer?

Reply:
Now I used `std::array` for this